pkg/xray/resource_xray_license_policy_test.go

```diff
@@ -0,0 +1,361 @@
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(fqrn, "name", policyName),
+					resource.TestCheckResourceAttr(fqrn, "description", policyDesc),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.name", ruleName),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.actions.0.build_failure_grace_period_in_days", strconv.Itoa(gracePeriod)),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.actions.0.custom_severity", "High"),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.actions.0.fail_build", "true"),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.actions.0.block_download.0.active", "false"),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.actions.0.block_download.0.unscanned", "false"),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.criteria.0.allowed_licenses.0", "Apache-1.0"),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.criteria.0.allowed_licenses.1", "Apache-2.0"),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.criteria.0.multi_license_permissive", "false"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckLicensePolicyDestroy(id string) func(*terraform.State) error {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[id]
+		if !ok {
+			return fmt.Errorf("error: Resource id [%s] not found", id)
+		}
+		provider, _ := testAccProviders["xray"]()
+		_, resp, _ := getPolicy(rs.Primary.ID, provider.Meta().(*resty.Client))
+
+		if resp.StatusCode() == http.StatusOK {
+			return fmt.Errorf("error: Policy %s still exists", rs.Primary.ID)
+		}
+		return nil
+	}
+}
+
+func testAccXrayLicensePolicy_badLicense(resourceName, name, description, ruleName string, rangeTo int) string {
+	return fmt.Sprintf(`
+resource "xray_security_policy" "%s" {
+	name = "%s"
+	description = "%s"
+	type = "license"
+	rules {
+		name = "%s"
+		priority = 1
+		criteria {
+			cvss_range {
+				from = 1
+				to = %d
+			}
+		}
+		actions {
+			block_download {
+				unscanned = true
+				active = true
+			}
+		}
+	}
+}
+`, resourceName, name, description, ruleName, rangeTo)
+}
+
+func testAccXrayLicensePolicy_badGracePeriod(resourceName, name, description, ruleName string, gracePeriod int) string {
+	return fmt.Sprintf(`
+resource "xray_license_policy" "%s" {
+	name = "%s"
+	description = "%s"
+	type = "license"
+	rules {
+		name = "%s"
+		priority = 1
+		criteria {
+		  allowed_licenses = ["Apache-1.0","Apache-2.0"]
+          allow_unknown = false
+          multi_license_permissive = true
+		}
+		actions {
+			fail_build = false
+			build_failure_grace_period_in_days = %d
+			block_download {
+				unscanned = true
+				active = true
+			}
+		}
+	}
+}
+`, resourceName, name, description, ruleName, gracePeriod)
+}
+
+func testAccXrayLicensePolicy_createAllowedLic(resourceName, name, description, ruleName string,
+	multiLicense string, gracePeriod int, blockUnscanned string, blockActive string) string {
+	return fmt.Sprintf(`
+resource "xray_license_policy" "%s" {
```

Comment:
When you have this many params, it's much clearer to use a template

Reply:
done